packages/engine/src/scene/components/WaterComponent.ts

```diff
@@ -43,6 +44,12 @@ export const WaterComponent = defineComponent({
     return water
   },
 
+  onSet: (entity, _component, json) => {
+    if (!json) return
+
+    setComponent(entity, ShadowComponent)
```

Comment:
`setComponent(entity, ShadowComponent)` should go before `if (!json) return` as well as in all other editors

Reply:
implemeted!